@types/pal/image.d.ts

```diff
@@ -0,0 +1,65 @@
+declare module 'pal/image' {
+    export type ImageSource = import('pal/image/types').ImageSource;
+    export type IMemoryImageSource = import('pal/image/types').IMemoryImageSource;
+    export type PixelFormat = import('cocos/asset/assets/asset-enum').PixelFormat;
+    export class ImageData {
+        constructor (imageAsset?: ImageSource | ArrayBufferView);
+        /**
+         * Destroy resources.
+         */
+        destroy (): void;
+
+        /**
+         * Get and set image data source.
+         * @param value Image data source.
+         */
+        get data(): ImageSource | ArrayBufferView;
+        set data(value: ImageSource | ArrayBufferView);
+
+        /**
+         * Get image width.
+         */
+        get width(): number;
+        /**
+         * Set image width(when source is IMemoryImageSource).
+         */
+        set width(value: number);
+
+        /**
+         * Get image height.
+         */
+        get height(): number;
+        /**
+         * Set image height(when source is IMemoryImageSource).
+         */
+        set height(value: number);
```

Comment:
It should not be combined here because the behavior of set is not the same

Reply:
It doesn't make sense to have separate getter/setter annotations, because it will only end up being parsed as one annotation by the TS language service